### DISCUSSION TITLE
Manual QGL doesn't fire events

### DISCUSSION THREAD

**Discussion by agreen** [jschuh/klipper-macros#155] [Ideas]
Not sure if this is really a bug or not, but manual QGL doesn't fire an event.  It looks like the reason why, is you aren't overriding the QUAD_GANTRY_LEVEL gcode with a custom macro which would then call the event. 

I had setup events to trigger my LED's when various things were going on, and was troubleshooting why my LED's weren't changing when I was doing QGL (manually) when I noticed the issue.

**Comment by jschuh** [comment]
I try to avoid overriding commands unless it's required for core functionality, although I suppose I could add it as an `optional` include. It should be as simple as something like this:

```
[gcode_macro quad_gantry_level]
rename_existing: _KM_QUAD_GANTRY_LEVEL
  _KM_PRINT_STATUS ACTION=PUSH_STATUS
  _KM_PRINT_STATUS ACTION=CHANGE STATUS=leveling_gantry
  _KM_QUAD_GANTRY_LEVEL {%rawparams%}
  _KM_PRINT_STATUS ACTION=CHANGE STATUS=pop_status
```

**Reply by agreen** [reply]
Yeah, I think that's a good point. It may be a worthwhile optional include like the couple of other optional includes you have.  There could be more events I haven't run across this could apply to also, and all of those collectively could be one include (or individual, whichever is more appropriate).
